python/src/connection.rs

```diff
@@ -85,55 +85,70 @@ impl Connection {
         future_into_py(self_.py(), async move { op.execute().await.infer_error() })
     }
 
+    #[pyo3(signature = (name, mode, data, storage_options = None))]
```

Comment:
Is this needed?  Is it for the default value?  I think, in other calls, we put defaults in the `.py` and don't bother with them in the `.rs`